webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -517,7 +515,12 @@ trait CachedFindRuleNodeStatusReports
     val nodeWithOutdatedCompliance = cache.filter {
       case (id, compliance) =>
         compliance.runInfo match {
-          case t: ExpiringStatus => t.expirationDateTime.isBefore(now)
+          // here, we have a special case for unexpected version: it is useless to recompute compliance until we don't have a new run,
+          // ie the node config was changed elsewhere. It means that "unexpected version" wins above "No report in interval",
+          // ie that that error is bigger.
+          case _: UnexpectedVersion => false
```

Comment:
this is for avoiding the loops for when a node is un "unexpected version". We don't have to compute again and again it's compliance, since it won't change until we get a new run.

Reply:
just to be sure, when policy are regenerated, then the compliance is invalidated and it's no longer unexpected but pending, right ?